src/gs-plugin-loader.c

```diff
@@ -3925,6 +3990,8 @@ gs_plugin_loader_init (GsPluginLoader *plugin_loader)
 	priv->auth_array = g_ptr_array_new_with_free_func ((GFreeFunc) g_object_unref);
 	priv->profile = as_profile_new ();
 	priv->settings = g_settings_new ("org.gnome.software");
+	g_signal_connect (priv->settings, "changed",
```

Comment:
If you connect to `changed::allow-updates` you can remove the check in the callback (but it's OK if upstream has it this way and you don't want to change it)